### PR TITLE
[MVP QA #1] Visual: spacings

### LINKED ISSUE #38202: [MVP QA #1] Visual: spacings
Hi there! I reviewed the latest build of the product editing experience and made some notes about changes we can make. In this issue, I outline improvements to typography, spacing, and other small visual details throughout the form.

**Note** This should also be tested with the StoreFront theme as it adds some styling to the blocks, see this fix for the summary block for example: https://github.com/woocommerce/woocommerce/pull/38251

**Acceptance criteria**

- [x] We should update the margin-left on the buttons in the top bar from 16px to 8px.
- [x] Spacing between the last field and the next sub section or between field groups and the "Advanced" button should be 32 pixels.
- [x] The padding around the subsections should be 32px 0 0 0.
- [ ] We need 4 px of space between a subsection's title and the description.
- [x] In the feedback form, can we fix the buttons at the bottom or prevent the modal from scrolling?
- [x] On mobile, the left and right body margins should be 32 px.
- [x] All modals should now have 8 px window radius.
- [x] The help icon in checkboxes small, detached from the item label, and vertically misaligned. Left margin of 4 px, no vertical adjustment; somehow it has a -2px set for the top margin).
- [x] The inner padding in the tab in the top bar should be 12px, not 16px.
- [x] In the edit product link modal, let's set this modal’s width to 650 px and remove the line separating the header from the contents.
- [x] When the product is in a draft state, the button in the top bar should say `Add`, not `Save`. It should change into `Update` once it's published on the store.

cc @pmcpinto

### PR DESCRIPTION
**Pull request by mdperez86**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #38202

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Got to `/wp-admin/tools.php?page=woocommerce-admin-test-helper`
2. Under Features tab make sure to enable `product-block-editor`
3. Then visit `/wp-admin/admin.php?page=wc-admin&path=/add-product`
4. The product form should not have the problem described in #38202

### CODE REVIEW

File: packages/js/product-editor/src/blocks/section/editor.scss
Comment (joshuatf) [inline packages/js/product-editor/src/blocks/section/editor.scss]: Nice! 🎉

File: packages/js/product-editor/src/blocks/section/block.json
Comment (joshuatf) [inline packages/js/product-editor/src/blocks/section/block.json]: We discussed this in our grooming call, but this is the only area the feels a little weird with the naming.

It would be great if we could match the gaps to our existing variable gap sizes (24px, 36px, 40px) and then match the naming there as well.  @jarekmorawski is that feasible or do these gaps need to be different than our current gaps?

**Review by joshuatf — COMMENTED** [review]
Looking great on my end!  Thanks for the additional clean-up with notices in this PR.

Just left one question we discussed before.  Let's see what @jarekmorawski says and we can update our naming accordingly.

File: packages/js/product-editor/src/blocks/section/block.json
Comment (jarekmorawski) [reply packages/js/product-editor/src/blocks/section/block.json]: Meaning that we'd change all the gaps between fields and sections to those you listed? I'd prefer to avoid that. The ones in Figma are consistent with WordPress (more or less) and balance white space and density well. For example, increasing the gap between sections from 32 px to 36 px or 40 px will impact how much content is visible before scrolling.

By the way, where do our current gap sizes come from and can we adjust them to the new ones?

Comment (joshuatf) [reply packages/js/product-editor/src/blocks/section/block.json]: Those gaps came from the previous designers, but I'm unsure on the specifics of how they were chosen.

It would be really great to adjust them, though that would require changes in many places (a quick scan shows that we're using some form of the `$gap` variable in 762 files).

But if this is something we plan on doing one day, we could at least update the naming in this PR to match those.  @jarekmorawski do you have ideal sizes for gaps/spacing for smallest, smaller, small, default, large, larger, and largest?

Comment (jarekmorawski) [reply packages/js/product-editor/src/blocks/section/block.json]: If we update the gaps, it'll affect all of wc-admin, correct? We'd have to do an audit to make sure we wouldn't break the visual rhythm outside the form, so I can't specify them yet. For the form, let's use the sizes below.

I'll leave it to your judgment and BEM best practices regarding naming.

- XXS: 8px
- XS: 16px
- S: 24px
- M: 32px
- L: 40px
- XL: 48px
- XXL: 56px

We'll unlikely ever use gaps larger than 56 px for form elements or cards like in the Home screen. The only things I can think of are margins and paddings, but I'm not sure we should include them in the system.

Comment (joshuatf) [reply packages/js/product-editor/src/blocks/section/block.json]: >If we update the gaps, it'll affect all of wc-admin, correct?

Correct, but we're not looking at tackling that in this PR.  Mostly just curious if it's something we'll eventually make sizes consistent between these and the existing gaps.  If so, we can match the naming and eventually replace them.

On the other hand, providing new sizes (xxs, xs, s, etc) provides us an easier way to transition over.  I'm indifferent between those sizes and using the previous semantics (smallest, smaller, small, etc).  @mdperez86 do you have an opinion on this?

Comment (mdperez86) [reply packages/js/product-editor/src/blocks/section/block.json]: @joshuatf in popular UI frameworks out there using (xs, sm, md, lg, xl) have become in a standard not only referring to a screen size but also to text-size and sometimes to spacing. But the problem this convention has is that the values are always fixed and we'll face the same problem in future if a new value appeared like now. Because of that some frameworks adopted a more scalable solution like using `gap-1`, `gap-2`, `gap-3` -> `0.25rem /* 4px */`, `0.5rem /* 8px */`, `0.75rem /* 12px */` which give us infinite flexibility if a new value is required like `0.375rem /* 6px */` that can be represented as `gap-1.5`. Of course the downside of this approach is that is more difficult to recognize than a more descriptive names like `xs` or `lg`.
So I'm ok using the convention suggested by @jarekmorawski but just consider we could face the same problem we have now in the future.

Comment (joshuatf) [reply packages/js/product-editor/src/blocks/section/block.json]: @mdperez86 I like that style better as well and I think WooCommerce is the only one tied to a system using our current style of naming.  I think that bodes well for us switching to a new relative gap system.

[Looks like WordPress has adopted a similar approach.](https://make.wordpress.org/design/2019/10/31/proposal-a-consistent-spacing-system-for-wordpress/). And luckily we already have access to those variables here:

https://github.com/WordPress/gutenberg/blob/b2c16f3c684098a8dabff3911c610caa5c053701/packages/base-styles/_variables.scss#L31-L41

Let's follow that convention and we can slowly deprecate our previous gap system in time.

Comment (mdperez86) [reply packages/js/product-editor/src/blocks/section/block.json]: done -> https://github.com/woocommerce/woocommerce/pull/38325/commits/ab0329f9cbd4ccf357a5338fdcac14e1fd8cccf6

File: packages/js/product-editor/src/blocks/section/edit.tsx
Comment (joshuatf) [inline packages/js/product-editor/src/blocks/section/edit.tsx:28]: Nice 👍

File: packages/js/product-editor/src/components/header/publish-button/publish-button.tsx
Comment (joshuatf) [inline packages/js/product-editor/src/components/header/publish-button/publish-button.tsx:39]: Noted this in the other PR, but won't let it block this PR since it's fixed there.

https://github.com/woocommerce/woocommerce/pull/38507#discussion_r1212336166

**Review by joshuatf — APPROVED** [review]
Thanks for all the hard work on this, Maikel!  This is a big improvement on our current spacing and how we organize things.

I noticed some oddities around spacing and background color in my environment, but I think this is unrelated to this PR and has to do with Gutenberg styles bleeding through since I have the latest GB enabled.  We can address that in a follow-up so we don't block this issue further.
Screen Shot 2023-06-01 at 10 28 29 AM

File: packages/js/customer-effort-score/src/style.scss
Comment (ilyasfoo) [inline packages/js/customer-effort-score/src/style.scss:110]: Hi, @mdperez86! It looks like `.components-modal__content` is way too general, it's affecting all modals in admin.

image

Originally reported by Vlad Olaru in this slack p1686918124623299/1686328973.559239-slack-C01SFMVEYAK.

Comment (octaedro) [reply packages/js/customer-effort-score/src/style.scss:110]: Hi @ilyasfoo, thank you for reporting that.
I created [this issue](https://github.com/woocommerce/woocommerce/issues/38748) and I'm working on a fix now.

**Pull request merged**
